katsdpimager/loader_katdal.py

```diff
@@ -222,15 +224,39 @@ def data_iter(self, start_channel, stop_channel, max_chunk_vis=None):
         # timestamps is a property, so ensure it's only evaluated once
         timestamps = self._file.timestamps
         baseline_idx = np.arange(len(self._baselines)).astype(np.int32)
-        for start in range(0, n_file_times, load_times):
+        start = 0
+        # Determine chunking scheme
+        if isinstance(self._file.vis, DaskLazyIndexer):
+            chunk_sizes = self._file.vis.dataset.chunks[0]
+            chunk_boundaries = [0] + list(itertools.accumulate(chunk_sizes))
```

Comment:
Why not `np.cumsum`?

Reply:
It didn't seem like something that needed to be taken to numpy-land, given that there are only a few thousand entries, and it didn't seem like numpy would make the code any more readable.